### PR TITLE
fix(payments): add Coupon Component to Subscription Success

### PR DESCRIPTION
**Pull request by IvoJP**
## Because

- We want to refllect coupon usage to the end user on the subscription success page

## This pull request

- adds state to the coupon component to make it suitable for the success page and adds it to said page
- disables input and buttons when we are trying to process a subscription

## Issue that this pull request solves

Closes: #11539 

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [X] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Review by StaberindeZA — CHANGES_REQUESTED** [review]
When I tested it locally I noticed that on the success screen, the Coupon Details wasn't showing up. I was using PayPal to complete the transaction. (Screenshot attached)

image

**Review by biancadanforth — COMMENTED** [review]
I know Reino already reviewed this, but @chenba and I took a pass together as well and left a few comments. One thing I noticed on the success page is the `CouponForm` is there, but disabled (i.e. `readOnly`) even when no coupon is used to subscribe. Is that what we want in this case? I would assume we wouldn't want the `CouponForm` to be in the document at all.

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx:41]: this existed before the interface did, so i copy and pasted the tests as they existed. i'll look into updating references.

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx:59]: i guess from a UI perspecitve they look the same but they are different states. i added them both to be exhaustive - if someone wanted to know what the coupon component would look like depending on where the user was at between clicking pay now vs being on the success page

File: packages/fxa-admin-server/src/graphql.ts
Comment (IvoJP) [reply packages/fxa-admin-server/src/graphql.ts]: added by linter when i pulled in main. reverted.

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx]: done

File: packages/fxa-payments-server/src/components/CouponForm/index.scss
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.scss:89]: it is, im removing it

**Comment by IvoJP** [comment]
@StaberindeZA i think you didnt see it in the off chance you pulled in the changes before main was merged in (although that was on thursday or friday) and before removing the feature flag was in main. 

if you pulled the latest you will not need to set any feature flags. i just did a test using paypal and it was on the success page
Screen Shot 2022-02-14 at 4 39 36 PM

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: changed to `checkingCoupon`

**Comment by IvoJP** [comment]
@biancadanforth there is not a figma for success but no coupon applied - https://www.figma.com/file/cOIAuHIIbrPbAlAIcpejNM/%F0%9F%8E%A1MVP-Subscription-Optimization-Flow?node-id=1532%3A17660 

but agreed i guess it doesnt make sense to show it if there is no coupon being used. i'll hide it.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx]: so from what i understood from the code `inProgress` is for stripe? and `transactionInProgress` is for paypal? the only place the latter is set to true is in the paypal button component. but when you have the paypal stuff open there is an overlay on the page that doesnt clear until the paypal transaction finishes.

i added it as a condition anyway so now its `inProgress || transactionInProgress` just for consistency/thoroughness

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx]: cleaned up

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx]: cleaned up

**Comment by IvoJP** [comment]
@biancadanforth

File: packages/fxa-payments-server/src/routes/Product/SubscriptionSuccess/index.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionSuccess/index.tsx]: added

File: packages/fxa-payments-server/src/components/CouponForm/index.test.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.test.tsx:59]: added

**Review by StaberindeZA — COMMENTED** [review]
LGTM, with 2 small changes. 

I also retested the PayPal scenario and worked for me. Thank you for the note.

I'll let @biancadanforth approve it after her re-review.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: updated

File: packages/fxa-payments-server/src/components/CouponForm/index.test.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.test.tsx:56]: done

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (biancadanforth) [inline packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx:8]: Nit: This isn't being used anymore, so can you remove it?

We have a VS Code extension called TS Hero that can do these things for you automatically e.g. on command or on save. It will also sort the imports alphabetically. From the Command Palette > TS Hero: Organize imports (sort and remove unused).

**Review by biancadanforth — APPROVED** [review]
LGTM thanks Ivo.

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx]: done

**Review by StaberindeZA — APPROVED** [review]
LGTM thank you.

**Pull request merged**
